### PR TITLE
Add subscript support

### PR DESCRIPTION
**Pull request by Tommytrg**
Add support for subscripts as arguments

### CODE REVIEW

File: dist/src/radon.js [hunk 1]
```diff
@@ -281,7 +281,13 @@ var Script = /** @class */ (function () {
         script.reduce(function (acc, item) {
             var op = new Operator(cache, _this.scriptId, acc, item, _this.onChildrenEvent());
             _this.operators.push(op);
-            return op.operatorInfo.outputType;
+            // if the outputType is same return its input type to be able to calculate the options
```
Comment (aesedepece) [inline dist/src/radon.js:284]:
```suggestion
            // If the `outputType` is `same` (a pseudo-type), return the input type so the available methods can be guessed correctly.
```

Comment (aesedepece) [reply dist/src/radon.js:284]: Yet this left me wondering... does `same` exist anymore?

File: src/radon.ts [hunk 2]
```diff
@@ -555,32 +562,60 @@ export class Operator {
...
+  public argument: Argument | Script | null
   public argumentInfo: ArgumentInfo
   public argumentType: MarkupArgumentType
   public cache: Cache
   public id: number
   public value: MirArgument | undefined
+  public subscript: boolean
 
-  constructor(cache: Cache, argumentInfo: ArgumentInfo, argument?: MirArgument) {
+  // TODO: find a better way to discrimate if the argument is a subscript, maybe adding a nested array
```
Comment (aesedepece) [inline src/radon.ts]:
```suggestion
  // TODO: find a better way to discriminate whether the argument is a subscript, maybe adding a nested array
```

**Review by aesedepece — CHANGES_REQUESTED** [review]

File: dist/src/radon.js (hunk 1 above)
Comment (Tommytrg) [reply dist/src/radon.js:284]: The output type for ArrayFilter and ArraySome operators is `same`

Comment (aesedepece) [reply dist/src/radon.js:284]: I agree that `ArrayFilter` must be `same` or `Array<inner>`, but shouldn't `ArraySome` outptut a `RadonBoolean`?

Comment (aesedepece) [reply dist/src/radon.js:284]: Anyway, you resolved by doubt :+1:

File: dist/src/radon.js [hunk 3]
```diff
@@ -412,21 +418,47 @@ var Operator = /** @class */ (function () {
...
+            name: EventName.Update,
+            data: { operator: { id: this.id, scriptId: this.scriptId } },
+        });
     };
     return Operator;
 }());
 exports.Operator = Operator;
 var Argument = /** @class */ (function () {
-    function Argument(cache, argumentInfo, argument) {
+    // TODO: find a better way to discrimate if the argument is a subscript, maybe adding a nested array
```
Comment (aesedepece) [inline dist/src/radon.js]:
```suggestion
    // TODO: find a better way to discriminate if the argument is a subscript, maybe adding a nested array
```

File: src/radon.ts [hunk 4]
```diff
@@ -608,6 +643,15 @@ export class Argument {
         },
       } as MarkupSelect
       // } else if (argumentType === MarkupArgumentType.SelectReduce) {
```
Comment (aesedepece) [inline src/radon.ts]: ???

**Review by aesedepece — APPROVED** [review]

**Pull request merged**